components/crud-web-apps/jupyter/README.md

```diff
@@ -68,26 +68,28 @@ make -C backend run-dev
 ```
 
 ### Internationalization
-Internationalization was implemented using [ngx-translate](https://github.com/ngx-translate/core).
-
-This is based on the browser's language. If the browser detects a language that is not implemented in the application, it will default to English.
-
-The i18n asset files are located under `frontend/src/assets/i18n`. One file is needed per language.
-
-The translation asset files are set in the `app.module.ts`, which should not be needed to modify.
-The translation default language is set in the `app.component.ts`.
-
-For each language added, `app.component.ts` will need to be updated.
-
-**When a language is added:** 
-- Copy the en.json file and rename is to the language you want to add. As it currently is, the culture should not be included.
-- Change the values to the translated ones
-
-**When a translation is added or modified:**
-- Choose an appropriate key
-- Make sure to add the key in every language file
-- If text is added/modified in the Common Project, it needs to be added/modified in the other applications as well.
+Support for non-English languages is only supported in a best effort way.
+
+Internationalization(i18n) was implemented using [Angular's i18n](https://angular.io/guide/i18n)
+guide and practices, in the frontend. You can use the following methods to
+ensure the text of the app will be localized:
+1. `i18n` attribute in html elements, if the node's text should be translated
+2. `i18n-{attribute}` in an html element, if the element's attribute should be
+   translated
+3. [$localize](https://angular.io/api/localize/init/$localize) to mark text in
+   TypeScript variables that should be translated
+
+The file for the English text is located under `i18n/messages.xlf` and other
+languages under their respective locale folder, i.e. `i18n/fr/messages.fr.xfl`.
+Each language's folder, aside from English, should have a distinct and up to
+date OWNERs file that reflects the maintainers of that language.
 
 **Testing**
 
-To test the i18n works as expected, simply change your browser's language to whichever language you want to test.  
+You can run a different translation of the app, locally, by running
+```bash
+ng serve --configuration=fr
```

Comment:
Do I understand correctly that the web apps will not dynamically change their language based on the browser language that is set after merging this PR?

/cc @wg102

Reply:
Correct. This will be the next step to discuss in the effort of using Angular's i18n approach [that's why I didn't close #6042 with this PR], which will require backend/manifests work as well https://angular.io/guide/i18n#deploy-multiple-locales.

Reply:
To avoid a regression the backend and manifest changes should be implemented before merging this PR.

Reply:
> To avoid a regression

This PR exists to fix regressions and bugs introduced from the initial PR that were not even brought up in the review, like 
1. The way the apps report errors https://github.com/kubeflow/kubeflow/issues/6041
2. Making the backend messages being readable only if they are parsed from the frontend
3. Locking other web apps [Katib, Models] to use `ngx-translate` in order to use the latest common library. Issues with `ngx-translate` are tracked here https://github.com/kubeflow/kubeflow/issues/6042

> the backend and manifest changes should be implemented before merging this PR.

This is bad practice especially for a huge work like internationalization that touches many parts of the app. This effort should happen in iterations by documenting each step along the way.

Reply:
I get your point though about the regression, since after the initial PR the apps had some internationalization support provided, and thank you for raising it. 

But I want to unblock upgrading the other web apps to use the latest common code, which right now means that they should follow the `ngx-translate` way. That's why I want to move forward with this first step, to untie the two efforts and be able to track them in parallel.

Reply:
The deployment and manifest changes needed to not cause a regression in these web apps will also be needed for the other web apps, and can be implemented completely independently from any upgrading or internationalization effort.

In fact, it is quite surprising a proper web server was never used to begin with, since it’s stated everywhere that you should use a web server like NGINX and not the use gunicorn to server the static files. 

Also, The other web apps haven’t been translated yet. As such, there is no reason they would need `ngx-translate`. The translation for those web apps can just as easily be done after the upgrade has been done, so this doesn’t block the upgrade in any way.

Reply:
I’d also like to point out that your statement `that were not even brought up in the review`, and other similar statements you’ve made, come across as blaming the reviewers of that PR for the bugs or regressions that slipped through the review process. Statements like this are appropriate within an open source project, not very welcoming for contributors and likely breach the community guidelines / code of conduct.

I remember a large PR being merged despite my review comments indicating a lot of issues in the code, but I’ll create a dedicated issue and PR to track the effort to fix this.

Reply:
> The other web apps haven’t been translated yet. As such, there is no reason they would need ngx-translate. The translation for those web apps can just as easily be done after the upgrade has been done, so this doesn’t block the upgrade in any way.

The common frontend code is using the `TranslateModule` from the `ngx-translate` module. I.e. main class for the [`BackendService`](https://github.com/kubeflow/kubeflow/blob/bf406cf03476a8238123b89957a8927ef0f53e43/components/crud-web-apps/common/frontend/kubeflow-common-lib/projects/kubeflow/src/lib/services/backend/backend.service.ts#L28) expects a `TranslateService` to be injected in order to work. 

This means that, in order to update the other apps to use the latest common code, they will need to install and use `ngx-translate`, which is something we want to avoid https://github.com/kubeflow/kubeflow/issues/6042.

Reply:
Still, the solution needed to avoid a regression can be implemented regardless of the use of `ngx-translate`.